src/Entities/DreamStuff/DreamDashCollider.cs

```diff
@@ -0,0 +1,116 @@
+        /// </summary>
+        /// <param name="player">The player instance.</param>
+        private bool Check(Player player) {
+            if (Active && Entity != null &&
+                player.GetData().Data.TryGetValue(Player_canEnterDreamDashCollider, out object canEnter) && canEnter.Equals(true)) {
+                
+                Collider collider = Entity.Collider;
+
+                Entity.Collider = Collider;
+                bool check = player.CollideCheck(Entity);
+                Entity.Collider = collider;
+
+                return check;
+            }
+            return false;
+        }
+
+        public override void Update() {
+            base.Update();
+            if (Util.TryGetPlayer(out Player player) && Check(player) && player.DashAttacking && player.Speed != Vector2.Zero && player.StateMachine.State != Player.StDreamDash) {
+                player.StateMachine.State = Player.StDreamDash;
+            }
+        }
+
+        public override void DebugRender(Camera camera) {
+            if (Collider != null) {
+                Collider collider = Entity.Collider;
+                
+                Entity.Collider = Collider;
+                Collider.Render(camera, Active ? ActiveColor : InactiveColor);
+                Entity.Collider = collider;
+            }
+        }
+
+        #region Hooks
+
+        public static readonly string Player_canEnterDreamDashCollider = "communalHelperCanEnterDreamDashCollider";
+
+        internal static void Load() {
+            IL.Celeste.Player.DreamDashUpdate += Player_DreamDashUpdate;
+            On.Celeste.Player.DashBegin += Player_DashBegin;
+            On.Celeste.Player.DreamDashEnd += Player_DreamDashEnd;
+        }
+
+        internal static void Unload() {
+            IL.Celeste.Player.DreamDashUpdate -= Player_DreamDashUpdate;
+            On.Celeste.Player.DashBegin -= Player_DashBegin;
+            On.Celeste.Player.DreamDashEnd -= Player_DreamDashEnd;
+        }
+
+        private static void Player_DashBegin(On.Celeste.Player.orig_DashBegin orig, Player self) {
+            orig(self);
+            self.GetData()[Player_canEnterDreamDashCollider] = true;
+        }
+
+        private static void Player_DreamDashEnd(On.Celeste.Player.orig_DreamDashEnd orig, Player self) {
+            DynData<Player> playerData = self.GetData();
+            if ((DreamBlock) playerData["dreamBlock"] is DreamBlockDummy dummy) {
+                foreach (DreamDashCollider collider in dummy.Entity.Components.GetAll<DreamDashCollider>()) {
+                    playerData[Player_canEnterDreamDashCollider] = false;
+                    collider.OnExit?.Invoke(self);
+                }
+            }
+            orig(self);
+        }
+
+        private static void Player_DreamDashUpdate(ILContext il) {
+            ILCursor cursor = new ILCursor(il);
+            if (cursor.TryGotoNext(MoveType.Before, instr => instr.MatchStloc(1))) {
```

Comment:
I'd like to match against something a little more reliable than just a `stloc`, is there a method or field or something near it that could be used as reference?